pkg/sdk/resource_monitors_test.go

```diff
@@ -0,0 +1,169 @@
+package sdk
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestResourceMonitorCreate(t *testing.T) {
+	id := randomAccountObjectIdentifier(t)
+
+	t.Run("empty options", func(t *testing.T) {
+		opts := &CreateResourceMonitorOptions{}
+		actual, err := structToSQL(opts)
+		require.NoError(t, err)
+		expected := "CREATE RESOURCE MONITOR"
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("with complete options", func(t *testing.T) {
+		creditQuota := Int(100)
+		frequency := FrequencyMonthly
+		startTimeStamp := "IMMIEDIATELY"
+		endTimeStamp := time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC).String()
+		notifiedUsers := []NotifiedUser{{Name: "FIRST_USER"}, {Name: "SECOND_USER"}}
+		triggers := []TriggerDefinition{
+			{
+				Threshold:     50,
+				TriggerAction: TriggerActionSuspendImmediate,
+			},
+			{
+				Threshold:     100,
+				TriggerAction: TriggerActionNotify,
+			},
+		}
+
+		opts := &CreateResourceMonitorOptions{
+			OrReplace:      Bool(true),
+			With:           Bool(true),
+			name:           id,
+			CreditQuota:    creditQuota,
+			Frequency:      &frequency,
+			StartTimeStamp: &startTimeStamp,
+			EndTimeStamp:   &endTimeStamp,
+			NotifyUsers:    &NotifyUsers{notifiedUsers},
+			Triggers:       &triggers,
+		}
+
+		actual, err := structToSQL(opts)
+		require.NoError(t, err)
+		expected := fmt.Sprintf(`CREATE OR REPLACE RESOURCE MONITOR %s WITH CREDIT_QUOTA = %d FREQUENCY = %s START_TIMESTAMP = '%s' END_TIMESTAMP = '%s' NOTIFY_USERS = ("FIRST_USER", "SECOND_USER") TRIGGERS ON %d PERCENT DO %s ON %d PERCENT DO %s`,
```

Comment:
not that its really an issue, but it may be easier to read to just have the expected be a hardcoded string instead of templated. Anyways the values are already known. It also saves you a bit of time because you can basically copy paste the result into expected instead of spending time to templating it.

Reply:
ok

Reply:
ok